src/packet/signature/types.rs

```diff
@@ -219,22 +222,23 @@ impl Signature {
     /// - when backsig is true: verify a "Primary Key Binding Signature (type ID 0x19)"
     fn verify_key_binding_internal(
         &self,
-        signing_key: &impl PublicKeyTrait,
-        key: &impl PublicKeyTrait,
+        signer: &impl PublicKeyTrait,
+        signee: &impl PublicKeyTrait,
         backsig: bool,
     ) -> Result<()> {
         debug!(
-            "verifying key binding: {:#?} - {:#?} - {:#?}",
-            self, signing_key, key
+            "verifying key binding: {:#?} - {:#?} - {:#?} (backsig: {})",
+            self, signer, signee, backsig
         );
 
-        let key_id = signing_key.key_id();
-        if let Some(issuer) = self.issuer() {
-            if &key_id != issuer {
+        // don't check issuer/issuer_fp subpacket match for backsigs. the subpackets are inconsistent in the test suite ("Binding signature subpackets"). (why?)
+        // just always verify is the signature itself is valid, regardless of issuer(_fp) subpackets.
+        if !backsig {
+            let key_id = signer.key_id();
+            if !Self::match_identity(self, signer) {
```

Comment:
I tend to think that this `match_identity` check should be entirely removed. A key binding, just by existing, always implicitly claims to be a self signature.

So I think key binding signatures should just be cryptographically checked, without taking issuer subpackets into account.

Reply:
Alright, i pushed a separate commit that drops this block.

If CI is happy, and we have consensus, I can squash that commit up.